### DISCUSSION TITLE
Allow to delete a discussion

### DISCUSSION THREAD

**Discussion by nesk** [community/community#2948] [Discussions]
I've just started [a discussion in phpstan/phpstan repository](https://github.com/phpstan/phpstan/discussions/4218) before realizing it was a duplicate… Not being proud of that (I've created the fifth discussion of  the repository and its a duplicate 👏), I wanted to delete the discussion, but it doesn't seem possible.

So I've written an apology and marked the discussion as _Answered_, however now I'm one of the _:airquote:_ most helpful _:airquote:_ people of the repository. 😅 

<img width="329" alt="image" src="https://user-images.githubusercontent.com/817508/101835345-ef5eaa80-3b3b-11eb-931f-219e0bf74a6e.png">

I understand you might not want to indefinitely allow deletion of discussions, but maybe we could be able to delete a discussion within the hour/day following its creation?

**Comment by WrichikBasu** [comment]
> I understand you might not want to indefinitely allow deletion of discussions, but maybe we could be able to delete a discussion within the hour/day following its creation?

Maybe we should be allowed to completely delete a discussion at least before someone else has posted an answer to it.

**Reply by nesk** [reply]
It would be perfect!

**Reply by nerdCopter** [reply]
100% agree
example:
* i've created a discussion with a stupid question including wrong information on the subject in question.
* i want to delete what i created, before a reply occurs, yet i cannot. :(

**Reply by saher2106** [reply]
yes we should have an option to delete discussions directly.

**Reply by Pacerier** [reply]
@WrichikBasu re "_has posted_"; Or has read. I certainly do not want something I have spent reading go missing.

**Comment by becca** [comment]
Similar to Issues, only repo admins and maintainers can delete an entire conversation on GitHub at this time. 
https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/deleting-an-issue

We can make this clearer in the docs.github.com/discussions content as we continue expanding on the documentation there.

**Reply by nesk** [reply]
I understand the current limitations. However, my question is rather: do you consider adding the ability to delete discussions by their authors or do you definitely reject the idea?

**Reply by nesk** [reply]
@becca Do you have an answer to this question?

> do you consider adding the ability to delete discussions by their authors or do you definitely reject the idea?

**Reply by becca** [reply]
@nesk all feedback submitted here is being considered but this specific request is not a top priority for us at the moment. are you asking for a delete functionality here for any other reasons beyond the `duplicate` problem stated in the OP? we're currently looking into how to better handle the duplicate problem in Discussions 😄

**Reply by nesk** [reply]
For the moment the only reason I can see is the duplicate one yes. Indeed, maybe the duplication matter can be handled another way. Thank you for your answer.

**Reply by dillingham** [reply]
@becca another reason: sometimes people figure out that their own question was an error on their part and not valuable to broadcast to the community. Withholding a delete option feels weird IMO, but probably has implications on notifications and such.

**Reply by Jakobud** [reply]
I think one of the big advantages of being able to delete your own discussion is this:

1. Someone is stuck on a problem in their code and assumes the library is bugged
2. They create a new discussion asking for help in understanding what is wrong with their code or for clarification on if there is a bug
3. They realize that nothing was wrong with their code or the library except for a simple typo that they fix and now everything works

So now, there is a discussion that is really not worth being there. It's not really helpful to anyone and won't help anyone solve any problems in the future. It's just another discussion that makes it more difficult for others in the future to find what they are looking for.

If that happened to me (and it has) I would immediately just delete the discussion, because what's the point of leaving it there?

Also, look at, basically every single type of forum implementation on any website. Users are able to delete their own forum threads. Same situation here.

**Reply by Jakobud** [reply]
Here's another use-case: User creates a discussion for a topic in the wrong repo. For example a user has a question about the React library. They click on the wrong browser tab or something and are on the Twitter Bootstrap repository page. They create a new discussion asking about their React questions.

Immediately after creating the discussion the user realizes they made it on the wrong repository. But they cannot delete it.

**Reply by tinwe** [reply]
I too find it a really weird choice, that the author aren't able to delete his own thread. Not even when there aren't any reactions or comments on it.

Quite often there are questions, ideas or requests, where the topic author finds the solutions to the question or request themselves. Or even being posted in the wrong repo.

I see no reason why the author shouldn't be able to delete his own thread. At least before there are no comments on it. It just hogs up discussions unecessarily.

**Reply by WASasquatch** [reply]
I think because it's a discussion section in general, like the standard in all forum software, the discussions should be the sole property of the author, because of liabilities. With only allowing admins to delete discussions, if even available, they are liable for the discussion. And with the high probability of an abandoned repo, that liability then falls on Github. Managing a discussion, including deleting the entire thread through base discussion post should be left to the author, extended to admins for moderation. 

Discussions could run fowl and have no administrating presence, and you can't even close, or delete your own topic.

**Reply by Pacerier** [reply]
@Jakobud re "_the point_"; point is that someone may need to refer to what they have read.

re "_more difficult for others_"; Nop. that's what **archiving** is for. which you can do using tags.

re "_every single type of forum_"; Nop. not for HN. not for Gmail (and all emails). not for books/journals/newspapers. etc

re "_But they cannot delete it_"; but they can reply \<Problem Solved> / \<Accidental Post> and archive it.

**Reply by Pacerier** [reply]
@tinwe re "_there aren't any reactions_"; yet if someone has spent as much as 380ms reading your thread, he could have started writing a reply.

re "_finds the solutions_"; then the **least** you can do is to reply so. With or without the solution.

re "_even_""_posted_""_wrong_"; ditto.

re "_no comments_";
> point is that someone may need to refer to what they have read

re "_hogs_""_unecessarily_"; neither unnecessary (re "_point_") nor hogs (re "___archiving___").

**Reply by RedCarlos26** [reply]
> I think because it's a discussion section in general, like the standard in all forum software, the discussions should be the sole property of the author, because of liabilities. With only allowing admins to delete discussions, if even available, they are liable for the discussion. And with the high probability of an abandoned repo, that liability then falls on Github. Managing a discussion, including deleting the entire thread through base discussion post should be left to the author, extended to admins for moderation. 
> 
> 
> 
> Discussions could run fowl and have no administrating presence, and you can't even close, or delete your own topic. 

Agreed. It baffles me that this option is still unavailable, 5 years later.

**Comment by Davekibh** [comment]
Can we Delete Discussion!

**Reply by becca** [reply]
Did my response here https://github.com/github/feedback/discussions/2948#discussioncomment-205422 not answer your question?

**Comment by PranamBhat** [comment]
I think once you mark it as "**Answered**", no need to delete. If you have a new discussion, you can always create one rather than editing the old one.

People can click on the filter "Answered" and "Unanswered" to have a look at their need.

Meanwhile this link is sufficient : [https://docs.github.com/en/github/managing-your-work-on-github/deleting-an-issue](https://docs.github.com/en/github/managing-your-work-on-github/deleting-an-issue)

**Reply by craigphicks** [reply]
That is a useful suggestion.

However I don't think it is sufficient for all cases. Answered questions are typically/hopefully very valuable because they have answers which will be helpful to others. In the case where the answer is probably not going to be a resource for others, allowing the OP to delete or hide the post is a beneficial filter, and it lessens the maintainers burden. (Yes, this assertion is debatable).

There is also the issue of privacy/security - if some private/secure information is accidentally published, the ability to immediately hide from anyone but the maintainers is critical. Not sure it is a "human right", but getting close.

**Reply by g-re-g** [reply]
Clearly OP being able to delete discussions that they have realized are not beneficial to the community is a win. If not least to cut down on having to filter out useless duplicates or questions that they quickly realized were unrelated to the repo. Marking them as answered clearly doesn't solve this problem and leads folks who see them marked as Answered to wonder why and if they're missing something.

**Reply by damien-roche** [reply]
Nope. I should have the right to remove my content. The end.

**Comment by tinwe** [comment]
I too find it a really weird choice, that the author aren't able to delete his own thread. Not even when there aren't any reactions or comments on it.

Quite often there are questions, ideas or requests, where the topic author finds the solutions to the question or request themselves. Or even being posted in the wrong repo.

I see no reason why the author shouldn't be able to delete his own thread. At least before there are no comments on it. It just hogs up discussions unecessarily.

**Comment by antonagestam** [comment]
Another usecase is releases apparently being able to create automatically create duplicate discussions for a release:

- https://github.com/antonagestam/phantom-types/discussions/182
- https://github.com/antonagestam/phantom-types/discussions/181

**Comment by reregaga** [comment]
I'll bump it up. I had an incident: I got a message from Github `This page is taking too long to load` when I created a question in Discussions and thought my question was not sent, so I sent it again, but it turned out that I duplicated my question and I would like to delete my duplicate. Here's an example of the use case.

**Comment by g-re-g** [comment]
This is clearly a reasonable feature and one that should be included in the discussion feature here. If there are no replies, allow the OP to delete the discussion. Often folks will post questions and then realize one of many reasons the discussion doesn't benefit the community.

* The answer is actually unrelated to where the discussion was posted
* The question is a duplicate
* The question is the result of a misunderstanding
* etc...

**Reply by damien-roche** [reply]
Keyword here being "clearly". The fact you can't delete your own discussions is absurd full stop, esp. those that haven't been answered.

**Comment by damien-roche** [comment]
You should OBVIOUSLY be able to delete your own unanswered discussions. What is the nonsensical reason for this? I'm finding it difficult to believe people are having to argue for a deletion feature.

All this will encourage is misuse. For eg, I can't delete, but I can edit to remove all content. Well done GH.

**Comment by lclarkg18** [comment]
This would be great, I just accidentally created two posts and found this while trying to find how to delete one of them

**Comment by LindseyB** [comment]
Thanks for your feedback. I've brought this to the team for discussion.

**Comment by kmk3** [comment]
(This comment is mostly a summary of this discussion)

Adding up some of the existing suggestions, the following seem like reasonable
requirements for deleting a discussion:

* The discussion was created [less than 1 day ago](https://github.com/orgs/community/discussions/2948#discussion-59817)
* No replies or answers [were posted yet](https://github.com/orgs/community/discussions/2948#discussioncomment-189184)
* Maintainers should still be able to delete it regardless of the above (this
  is currently the case)

These should help prevent old discussions with useful information being deleted
without warning (due to hacked accounts, etc).

It seems like it could cover the following in most cases:

* A discussion was accidentally [submitted twice](https://github.com/orgs/community/discussions/2948#discussioncomment-5178641)
* A discussion was opened [in the wrong project](https://github.com/orgs/community/discussions/2948#discussioncomment-1924060)
* The solution was found [shortly after the discussion was opened](https://github.com/orgs/community/discussions/2948#discussioncomment-1924060)

Lastly, note that the following actions are already possible:

* Users can edit their own comments [and delete previous versions](https://github.com/orgs/community/discussions/2948#discussioncomment-4941397)
* Discussions can be closed as duplicates (closed discussions are hidden by
  the default)
* Maintainers can [delete discussions](https://github.com/orgs/community/discussions/2948#discussioncomment-205422)

**Comment by kmk3** [comment]
Implementation-wise, if the discussion cannot be deleted, I'd suggest greying
out the `[Delete this discussion]` button and explaining the reason(s) when
hovering over it:

> Discussions older than _1 day_ cannot be deleted

Or:

> Discussions containing replies cannot be deleted

With possibly a `(?)` button on the right, which links to the documentation (I
think I've seen this before on GitHub, but can't remember where):

* `[Delete this discussion] (?)`

These would make it clear to the user why a discussion cannot be deleted.
Currently the button is simply hidden when the user does not have the
permissions required to delete a discussion.

**Comment by tinwe** [comment]
Great summary. Can we get a reply from a team member perhaps?

**Comment by LindseyB** [comment]
Unfortunately, this isn't something that is going to be changed at this point.

**Reply by tinwe** [reply]
> Unfortunately, this isn't something that is going to be changed at this point. 

Why?

**Reply by LindseyB** [reply]
Because of this https://github.com/orgs/community/discussions/2948#discussioncomment-205422

**Reply by tinwe** [reply]
> Because of this https://github.com/orgs/community/discussions/2948#discussioncomment-205422 

Becca just stated that it wasn't their top priority two years ago and asked what other reasons except duplicate, would warrant a need for deletion. And she got plenty of replies, so I frankly don't see a reason for why it wouldn't be developed eventually.
Still would prefer an updated reply from the team however.

**Reply by LindseyB** [reply]
I'm from the team. I can't go into more details without exposing some internal things that I cannot do, but it will not be prioritized or worked on. 

Unfortunately, that's as specific as I can be and I understand that this is frustrating, but my hands are tied.

**Reply by WASasquatch** [reply]
What's weird to me, is how Github doesn't understand this is a legal liability concern. Leaving it to repo admins makes them liable for content of discussions. I don't know a single discussion software (forums) that doesn't ship with a terms of use that doesn't immediately, and explicitly relinquish any ownership or holdings of users posts, and leave it explicitly with the author. There isn't even any legal terms of use regarding the discussion sections of repos, just a "guide" which isn't proper documentation. This isn't good for project owners.

**Reply by n3rada** [reply]
I am also pretty sure than the [GDPR](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=CELEX%3A32016R0679) is not well respected due to the right for individuals to have personal data erased. Not allowing someone to delete it's own content does not seems right. What's your opinion @LindseyB?

**Comment by tinwe** [comment]
> I am also pretty sure than the GDPR is not well respected due to the right for individuals to have personal data erased. Not allowing someone to delete it's own content does not seems right. What's your opinion @LindseyB?

That is certainly a valid argument as well. If we ignore the whole unecessary spam argument for when one obviously makes a post or double post by misstake, and need to delete it even before anyone has read, never the less replied, to the comment, the whole GDPR even makes it an important legal matter, to be able to delete your own comments from public view.

**Comment by WASasquatch** [comment]
At this point it will have to take a class-action lawsuit from us users against GitHub to provide transparency in regards to discussions with proper terms of use that exposes why they will not let us have standard control of our content, or force them to relinquish control of our content. 

And I am straight serious. This isn't something to camp on make vague sweeping verdicts on and is a serious issue. 

According to CCPA, VCDPA, CRPA, GRPR, the only exception to these rules is if it makes it impossible to do their research on the data. So what research is being carried out on our data. AI troubleshooting bot GitHub hopes to release? What excuse is there. It seems to actually have to be transparent and not "we can't say" like Lindsay suggests. There is one valid exception they can use here from my understanding of these laws, and it is being executed with gross negligence. 

To me it seems clear this isn't some sort of research firm where these exceptions would even apply. It's not like exposure could reach people to make everyone opt out immediately rendering the Research unable to proceed.

**Comment by binaryfire** [comment]
We should at least be able to delete discussions that haven't had any replies yet. That would allow users who have posted in the wrong repo etc. to correct their mistake. I can't see any logical reason to prevent that.

**Reply by tinwe** [reply]
> We should at least be able to delete discussions that haven't had any replies yet. That would allow users who have posted in the wrong repo etc. to correct their mistake. I can't see any logical reason to prevent that.

Hear! Hear!

**Reply by reii23** [reply]
I agree, they could be notified as errors and if any admin agrees, delete them.

**Comment by PavanLuca** [comment]
You might be interested in my discussion 👉 https://github.com/orgs/community/discussions/128602.

**Comment by 411A** [comment]
There are plenty of logical reasons to allow the deletion of empty discussions. It only takes 10 minutes of front-end work and 2 hours on the back-end to implement this feature... yet no action has been taken after 4 years. Interesting.

**Comment by n3rada** [comment]
I'm sure it won't be necessary to involve an extra layer of lawyers, GitHub is mature enough to understand that GDPR fines aren't something that can be passed up easily. Isn't that right @LindseyB?👀

**Reply by letanloc1998** [reply]
We might have to flood GitHub with discussions containing racist content before they finally implement a feature to delete discussions.

I’m concerned if someone accidentally posts personal information or company-sensitive data, they’d have no way to remove it.

**Comment by letanloc1998** [comment]
Oh no, I post sensitive data with my org account and cannot delete
